src/services/rfqm_service.ts

```diff
@@ -789,6 +817,92 @@ export class RfqmService {
         return submissionsMap;
     }
 
+    /**
+     * Recover context from a previous submission attempt
+     */
+    private async _recoverSubmissionContextAsync(
+        workerAddress: string,
+        orderHash: string,
+        callData: string,
+        previousSubmissions: RfqmTransactionSubmissionEntity[],
+    ): Promise<SubmissionContext> {
+        logger.info({ workerAddress, orderHash }, `previous submissions found, recovering context`);
+        const submissionsMap: SubmissionsMap = {};
+
+        // setting values to override them
+        const nonce = previousSubmissions[0].nonce;
+        let gasPrice = previousSubmissions[0].gasPrice;
```

Comment:
If you add the exclamation marks here, does it get rid of the need for them later?
```suggestion
        const nonce = previousSubmissions[0].nonce!;
        let gasPrice = previousSubmissions[0].gasPrice!;
```

Reply:
it does not